### ISSUE TITLE
Loss converged but no detections?

### ISSUE THREAD

**Issue by rremani** [rbgirshick/py-faster-rcnn#535]
Hi,
I just used one image with like 190 annotations for localizing characters from a form. Loss started with 10 and after 900 iterations loss was 0.01, after 10k iterations loss was 1e-4, but when I used the model to detect from the same image I got zero detection. 

Does anybody have idea what could be the reason. Image size was height 1000 and width 600.

**Comment by rremani** [comment]
@Dectinc @wangdelp  To detect 191 objects of same class from a single image training...shouldn't it be overfit but why I am not getting even single detection?

**Comment by jinyu121** [comment]
Hi～ I also met this problem several days ago, and I worked it out just now.

Please check your annotations, look carefully if any box out of image range. For example, x_min < 0, or the box is much larger than the image, or the box is totally out of image. 

Here is a script to [check the boxes](https://gist.github.com/jinyu121/2fb3264331241a3ec66039838ab686f6#file-range_check-py),and you can also use the [LabelImg tool](https://github.com/jinyu121/labelImg) to check them.

I fixed some annotations, and it works well now.

**Comment by rremani** [comment]
Hi, Thanks a lot but my annotations are fine. So there isn't any problem with number of predictions in a single image? Also my image was binary.

**Comment by djdam** [comment]
be careful: you have to check the width and height of the boxes AFTER resize. In my case, the original boxes in the images were fine, but there were 2 boxes with zero width/height in the roidb.
Here is a script I use for imdb analysis:

https://github.com/djdam/faster-rcnn-scenarios/blob/master/src/analysis/imdb_analyse.py

(you have to add some fields to your custom imdb script first:

{'boxes' : boxes,
                'filename': filename,
                'gt_classes': gt_classes,
                'gt_overlaps' : overlaps,
                'flipped' : False,
                'seg_areas' : seg_areas,
                'width': width,
                'height': height}

**Comment by rremani** [comment]
Thanks @djdam.

**Comment by djdam** [comment]
@rremani was that the problem?

**Comment by rremani** [comment]
@djdam Is there a limit in faster-rcnn for total number of detections in an image? there wasn't any such problem with annotations as they were in the middle of the image.

**Comment by djdam** [comment]
I don't think so, but I am not sure to be honest. I know that you can configure the "batch size" for both RPN as well as Fast RCNN parts, which basically limits the number of proposals fed to the RCNN part of the network.

**Comment by rremani** [comment]
ohk, thanks @djdam ill look more into it.

**Comment by cy94** [comment]
There is an issue with large images not training/testing with cudnn < v5. Try this out https://github.com/rbgirshick/py-faster-rcnn/issues/237
